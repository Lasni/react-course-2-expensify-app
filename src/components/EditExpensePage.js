import React from 'react';
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'


export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.startEditExpense(this.props.expense.id, expense)
    this.props.history.push('/')
  }

  onRemove = () => {
    this.props.startRemoveExpense({ id: this.props.expense.id })
    this.props.history.push('/')
  }

  render() {
    return (
      <div>
        <ExpenseForm
          expense={this.props.expense}
          onSubmit={this.onSubmit}
        />

        <button onClick={this.onRemove}>
          Remove expense
        </button>
      </div>
    )
  }
}


// const EditExpensePage = (props) => {
//   console.log(props)
//   return (
//     <div>
//       <ExpenseForm
//         expense={props.expense}
//         onSubmit={(expense) => {
//           props.dispatch(editExpense(props.expense.id, expense))
//           props.history.push('/')
//         }}
//       />

//       <button onClick={() => {
//         props.dispatch(removeExpense({ id: props.expense.id }))
//         props.history.push('/')
//       }}>
//         Remove expense
//       </button>
//     </div>
//   );
// };

const mapStateToProps = (state, props) => ({
  expense: state.expenses.find((expense) => expense.id === props.match.params.id)
})

const mapDispatchToProps = (dispatch, props) => ({
  startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
  startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
})

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);