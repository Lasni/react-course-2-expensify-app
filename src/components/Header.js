import React from 'react';
// import { NavLink } from 'react-router-dom';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth'

export const Header = ({ startLogout }) => (
  <header className='header'>
    <div className='content-container'>
      <div className='header__content'>
        <Link className='header__title' to='/dashboard'>
          <h1>Expensify</h1>
        </Link>
        {/*<NavLink to='/create' activeClassName='is-active'>Create expense</NavLink>*/}
        <button className='button button--link' onClick={startLogout}>Logout</button>
      </div>
    </div>
  </header>
)

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);