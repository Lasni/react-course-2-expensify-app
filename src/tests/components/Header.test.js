// import ReactShallowRenderer from 'react-test-renderer/shallow'
import React from 'react'
import { Header } from '../../components/Header'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'


test('should render header correctly', () => {
  // const renderer = new ReactShallowRenderer()
  // renderer.render(<Header/>)
  // expect(renderer.getRenderOutput()).toMatchSnapshot()

  const wrapper = shallow(<Header startLogout={() => {}}/>)
  expect(wrapper).toMatchSnapshot()

  // expect(wrapper.find('h1').text()).toBe('Expensify')
})

test('should call startLogout on button click', () => {
  const startLogout = jest.fn()
  const wrapper = shallow(<Header startLogout={startLogout} />)
  wrapper.find('button').simulate('click')
  expect(startLogout).toHaveBeenCalled()
})
