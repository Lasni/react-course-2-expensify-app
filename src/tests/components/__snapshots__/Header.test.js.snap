// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    exact={true}
    to="/"
  >
    dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Create expense
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</header>
`;
